pkg/storage/gc/policy.go

```diff
@@ -0,0 +1,198 @@
+package gc
+
+import (
+	glob "github.com/bmatcuk/doublestar/v4"
+	ispec "github.com/opencontainers/image-spec/specs-go/v1"
+
+	zerr "zotregistry.io/zot/errors"
+	"zotregistry.io/zot/pkg/api/config"
+	zcommon "zotregistry.io/zot/pkg/common"
+	zlog "zotregistry.io/zot/pkg/log"
+	mTypes "zotregistry.io/zot/pkg/meta/types"
+	"zotregistry.io/zot/pkg/storage/gc/retention/candidate"
+	"zotregistry.io/zot/pkg/storage/gc/retention/rules"
+	"zotregistry.io/zot/pkg/storage/gc/retention/rules/dayspull"
+	"zotregistry.io/zot/pkg/storage/gc/retention/rules/dayspush"
+	"zotregistry.io/zot/pkg/storage/gc/retention/rules/latestpull"
+	"zotregistry.io/zot/pkg/storage/gc/retention/rules/latestpush"
+	"zotregistry.io/zot/pkg/storage/gc/retention/rules/name"
+	"zotregistry.io/zot/pkg/storage/gc/retention/rules/retainall"
+)
+
+type policyManager struct {
+	repoPolicies []config.GCPolicy
+	log          zlog.Logger
+}
+
+func (p policyManager) hasDeleteUntagged(repo string) bool {
+	if policy, err := p.getRepoPolicy(repo); err == nil {
+		return policy.DeleteUntagged
+	}
+
+	// default
+	return false
+}
+
+func (p policyManager) hasDeleteReferrer(repo string) bool {
+	if policy, err := p.getRepoPolicy(repo); err == nil {
+		return policy.DeleteReferrers
+	}
+
+	// default
+	return false
+}
+
+func (p policyManager) hasTagRetention(repo string) bool {
+	if policy, err := p.getRepoPolicy(repo); err == nil {
+		return len(policy.TagsRetention) > 0
+	}
+
+	// default
+	return false
+}
+
+func (p policyManager) getNameRule(repo, tag string) rules.Rule {
+	repoPolicy, err := p.getRepoPolicy(repo)
+	if err != nil {
+		// if repo policy not found, then retain all
+		return retainall.New(true)
+	}
+
+	tagPolicy, err := p.getTagPolicy(tag, repoPolicy.TagsRetention)
+	if err != nil {
+		// if repo policy found but tag not found, then remove all tags
+		return retainall.New(false)
+	}
+
+	// name rule
+	return name.New(tagPolicy.Names)
+}
+
+func (p policyManager) getRules(repo string, tag string) []rules.Rule {
+	rules := make([]rules.Rule, 0)
+
+	repoPolicy, err := p.getRepoPolicy(repo)
+	if err != nil {
+		// if repo policy not found, then do not apply any tag retention policy
+		return rules
+	}
+
+	tagPolicy, err := p.getTagPolicy(tag, repoPolicy.TagsRetention)
+	if err != nil {
+		// if tag policy not found, gc all tags by default
```

Comment:
```suggestion
		// if tag policy not found, retain all tags by default
```

Correct? The specified tags in the config are the ones we want to keep, not the ones to delete, right?